settings.go

```diff
@@ -1,46 +1,201 @@
 package desktop
 
 import (
+	wmtheme "fyne.io/desktop/theme"
+	"fyne.io/fyne/canvas"
+	"fyne.io/fyne/cmd/fyne_settings/settings"
+	"fyne.io/fyne/layout"
+	"fyne.io/fyne/theme"
+	"fyne.io/fyne/widget"
 	"os"
+	"os/exec"
+	"strings"
 
 	"fyne.io/fyne"
 )
 
 // DeskSettings describes the configuration options available for Fyne desktop
 type DeskSettings interface {
-	IconTheme() string
 	Background() string
+	IconTheme() string
+	DefaultApps() []string
 }
 
 type deskSettings struct {
-	iconTheme  string
-	background string
+	background  string
+	iconTheme   string
+	defaultApps []string
+}
+
+const randrHelper = "arandr"
+
+func (d *deskSettings) Background() string {
+	return d.background
 }
 
 func (d *deskSettings) IconTheme() string {
 	return d.iconTheme
 }
 
-func (d *deskSettings) Background() string {
-	return d.background
+func (d *deskSettings) DefaultApps() []string {
+	return d.defaultApps
+}
+
+func (d *deskSettings) setBackground(name string) {
+	d.background = name
+	fyne.CurrentApp().Preferences().SetString("background", d.background)
+	canvas.Refresh(Instance().(*deskLayout).container)
+}
+
+func (d *deskSettings) setIconTheme(name string) {
+	d.iconTheme = name
+	fyne.CurrentApp().Preferences().SetString("icontheme", d.iconTheme)
+	canvas.Refresh(Instance().(*deskLayout).container)
+}
+
+func (d *deskSettings) setDefaultApps(defaultApps []string) {
+	newDefaultApps := strings.Join(defaultApps, "|")
+	fyne.CurrentApp().Preferences().SetString("defaultapps", newDefaultApps)
 }
 
 func (d *deskSettings) load() {
-	env := os.Getenv("FYNEDESK_ICONTHEME")
-	if env == "" {
-		d.iconTheme = "hicolor"
+	env := os.Getenv("FYNEDESK_BACKGROUND")
+	if env != "" {
+		d.background = env
 	} else {
-		d.iconTheme = env
+		d.background = fyne.CurrentApp().Preferences().String("background")
 	}
 
-	env = os.Getenv("FYNEDESK_BACKGROUND")
+	env = os.Getenv("FYNEDESK_ICONTHEME")
 	if env != "" {
-		d.background = env
+		d.iconTheme = env
 	} else {
-		d.background = fyne.CurrentApp().Preferences().String("background")
+		d.iconTheme = fyne.CurrentApp().Preferences().String("icontheme")
```

Comment:
Should we not fall back to "hicolor" as before if there is no setting specified?